ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -1010,7 +1010,11 @@ public boolean restoreState(InputObjectState os)
 
 	                    // Give it a go ourselves
 	                    if (!deserialized) {
-	                        _theXAResource = (XAResource) o.readObject();
+	                        try {
+	                            _theXAResource = (XAResource) o.readObject();
+	                        } catch (ClassNotFoundException cnfe) {
```

Comment:
It looks like better to move the below jtaxLogger.logger.trace("...") into the try-catch clause.